### PR TITLE
automated scenario for service specified experiment

### PR DESCRIPTION
**Pull request by kalantar**
automated e2e scenario for v0.2 -- service centric experiment specification. Addresses issue https://github.com/iter8-tools/iter8-controller/issues/180.

### CODE REVIEW

**Comment by kalantar** [comment]
This will probably fail testing. It is written in the expectation of upcoming controller changes and is dependent on them. See issue https://github.com/iter8-tools/iter8-controller/issues/156.

**Comment by kalantar** [comment]
@sriumcp this may be enough to allow you to get started on issue https://github.com/iter8-tools/docs/issues/90

File: doc/tutorials/istio/bookinfo/service/canary_productpage-v1_to_productpage-v2.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+apiVersion: iter8.tools/v1alpha1
+kind: Experiment
+metadata:
+  name: productpage-v2-rollout
+spec:
+  routingReference:
+    apiVersion: networking.istio.io/v1alpha3
+    kind: VirtualService
+    name: bookinfo
```
Comment (kimikowang) [inline doc/tutorials/istio/bookinfo/service/canary_productpage-v1_to_productpage-v2.yaml:9]: The vs referenced here does not exist.

**Comment by kimikowang** [comment]
What is the reason of creating all the new service/deployments yaml?

**Comment by kimikowang** [comment]
And this does not conform to the design doc use case. Please take a look at the targeted user case.

File: doc/tutorials/istio/bookinfo/service/canary_productpage-v1_to_productpage-v2.yaml (hunk 1 above)
Comment (kalantar) [reply doc/tutorials/istio/bookinfo/service/canary_productpage-v1_to_productpage-v2.yaml]: Agree. My mistake. I've removed this,

**Comment by kalantar** [comment]
> What is the reason of creating all the new service/deployments yaml?

I believe we need to redefine our definition to have different services for each new version and a new gateway definition. For simplicity of the user, I thought it made sense to include all the necessary pieces together. Otherwise, I think we would need to break up the bookinfo-tutorial.yaml file.

**Comment by kalantar** [comment]
> What is the reason of creating all the new service/deployments yaml?

@kimikowang After our discussion, I removed the duplication of the yaml and have just replaced where needed and added additional.

File: test/e2e/e2e-scenario-5s.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+kubectl apply -n $NAMESPACE -f $DIR/../../doc/tutorials/istio/bookinfo/service/canary_productpage-v1_to_productpage-v2.yaml
+kubectl get experiments -n $NAMESPACE
+
+header "Deploy canary version"
+kubectl apply -n $NAMESPACE -f $DIR/../../doc/tutorials/istio/bookinfo/productpage-v2.yaml \
+              -f $DIR/../../doc/tutorials/istio/bookinfo/service/productpage-v2.yaml
+sleep 1
+kubectl wait --for=condition=ExperimentCompleted -n $NAMESPACE experiments.iter8.tools productpage-v2-rollout --timeout=600s
+kubectl get experiments -n $NAMESPACE
+kubectl get vs bookinfo -n $NAMESPACE -o yaml
```
Comment (kimikowang) [inline test/e2e/e2e-scenario-5s.sh:47]: This should be removed, or `kubectl get vs -n $NAMESPACE -o yaml`.

**Pull request merged**